README.md

```diff
@@ -72,6 +81,18 @@ On the prometheus side you can set a scrape config as follows
             - target_label: __address__
               # The mysqld_exporter host:port
               replacement: localhost:9104
+              
+##### Socket authentication
+
+To use passwordless socket authentication, give socket based grants as documented above, then use:
+
+```shell
+## run as user mysqld_exporter:
+export DATA_SOURCE_NAME="mysqld_exporter:@unix(/var/run/mysqld/mysqld.sock)/"
```

Comment:
Thanks, but we need to update this to use the new config file format or command line flag (`--mysqld.address=`).

Reply:
How would the new format look like?

i tried both, but it doesn't seem to work:

```
--mysqld.address="unix(/var/run/mariadb/mysql.sock)"

2023-07-05T16:16:30.833935+02:00 db mariadb_exporter[3181430]: ts=2023-07-05T14:16:30.832Z caller=mysqld_exporter.go:225 level=info msg="Error parsing host config" file=.my.cnf err="failed to parse address: address unix(/var/run/mariadb/mysql.sock): missing port in address"
```

```
--mysqld.address="unix:///var/run/mariadb/mysql.sock"

2023-07-05T16:17:31.077223+02:00 db mariadb_exporter[3181808]: ts=2023-07-05T14:17:31.075Z caller=config.go:146 level=error msg="failed to parse config" section=client err="set field \"port\": strconv.ParseInt: parsing \"///var/run/mariadb/mysql.sock\": invalid syntax"
```